src/api/app/controllers/webui2/package_controller.rb

```diff
@@ -27,18 +27,21 @@ def webui2_statistics
   end
 
   def webui2_prefill
+    target_project = ''
+    target_package = ''
+    description = ''
+    cleanup_source = @project.branch? # We should remove the package if this request is a branch
     linked_package = @package.backend_package.links_to
     if linked_package
       target_project = linked_package.project.name
       target_package = linked_package.name
       description = @package.commit_message(target_project, target_package)
     end
-    cleanup_source = @project.branch? # We should remove the package if this request is a branch
 
     render json: {
-      'targetProject': target_project ||= "",
-      'targetPackage': target_package ||= "",
-      'description': description ||= "",
+      'targetProject': target_project,
+      'targetPackage': target_package,
+      'description': description,
```

Comment:
I just fail to see why these temporary variable is better than call ` linked_package.project.name`,  `linked_package.name` and `@package.commit_message(target_project, target_package)`  direct in the render?

Reply:
Because using `linked_package.project.name` and friends should only be done when there is actually a `linked_package`. I could move the render inside the if and provide an else branch with another render with blank values, but then I'll be repeating the render block. It's a matter of taste which version to use I guess as both versions have pros and cons.

Reply:
yes, sure. But I think there are better ways to achieve it, other than initialize an empty string. Either you encapsulate this logic in method or render what you have and make sure in the view you validate if the data exists before to try to render it.

i.e: Add keys to a hash and render the hash as it is, and in the view you do a `hash.fetch(:key, $DEFAULT_VALUE)`

Reply:
We've found some other way to refactor this without initializing the variables. Will this work for you @vpereira ?